tests/rewards/test_reward_nets.py

```diff
@@ -168,21 +196,27 @@ class WrapperB(reward_nets.RewardNetWrapper):
         def forward(*args):
             ...  # pragma: no cover
 
-    reward_net = RewardNetA(None, None)  # This might fail if we setup better input
-    # validation
+    reward_net = RewardNetA(env.action_space, env.observation_space)
     reward_net = WrapperB(reward_net)
 
     assert isinstance(reward_net.base, RewardNetA)
 
     # This should not raise a type error
-    serialize._validate_wrapper_structure(reward_net, [[WrapperB, RewardNetA]])
+    serialize._validate_wrapper_structure(reward_net, {(WrapperB, RewardNetA)})
 
     # The top level wrapper is an instance of WrapperB this should raise a type error
     with pytest.raises(
         TypeError,
-        match=r"Wrapper structure should be match \(one of\) \[.*\] but found \[.*\]",
+        match=r"Wrapper structure should match \[.*\] but found \[.*\]",
     ):
-        serialize._validate_wrapper_structure(reward_net, [[RewardNetA]])
+        serialize._validate_wrapper_structure(reward_net, {(RewardNetA,)})
+
+    # Reward net is not wrapped at all this should raise a type error.
+    with pytest.raises(TypeError):
```

Comment:
I notice you drop the `match=` keyword argument here. If that's just for brevity, you could define earlier on:

```
raise_ctx_mgr = pytest.raises(TypeError, match=...)
```

and then do `with raise_ctx_mgr` each time to avoid duplication.

(I think this works; if not, you could definitely make this work using functools.partial.)